aptos-move/block-executor/src/executor.rs

```diff
@@ -1043,6 +1161,233 @@ where
+                        &self.config.onchain.block_gas_limit_type,
+                        scheduler_wrapper,
+                        versioned_cache,
+                        last_input_output,
+                        shared_sync_params.block_limit_processor,
+                        base_view,
+                        global_module_cache,
+                        runtime_environment,
+                        start_delayed_field_id_counter,
+                        delayed_field_id_counter,
+                        &executor,
+                        block,
+                        num_workers as usize,
+                    )?;
+                }
+
+                scheduler.commit_hooks_unlock();
+            }
+
+            // TODO(BlockSTMv2): pass worker_id to next_task.
+            match scheduler.next_task()? {
+                TaskKind::Execute(txn_idx, incarnation) => {
+                    if incarnation > num_workers.pow(2) + num_txns + 10 {
+                        // Something is wrong if we observe high incarnations (e.g. a bug
+                        // might manifest as an execution-invalidation cycle). Break out
+                        // to fallback to sequential execution.
+                        error!("Observed incarnation {} of txn {txn_idx}", incarnation);
+                        return Err(PanicOr::Or(ParallelBlockExecutionError::IncarnationTooHigh));
+                    }
+
+                    Self::execute_v2(
+                        txn_idx,
+                        incarnation,
+                        block,
+                        last_input_output,
+                        versioned_cache,
+                        &executor,
+                        base_view,
+                        shared_sync_params.global_module_cache,
+                        runtime_environment,
+                        ParallelState::new(
+                            versioned_cache,
+                            scheduler_wrapper,
+                            start_delayed_field_id_counter,
+                            delayed_field_id_counter,
+                            incarnation,
+                        ),
+                        scheduler,
+                    )?;
+                },
+                TaskKind::PostCommitProcessing(txn_idx) => {
+                    self.materialize_txn_commit(
+                        txn_idx,
+                        versioned_cache,
+                        scheduler_wrapper,
+                        start_delayed_field_id_counter,
+                        delayed_field_id_counter,
+                        last_input_output,
+                        base_view,
+                        shared_sync_params.global_module_cache,
+                        runtime_environment,
+                        shared_sync_params.final_results,
+                    )?;
+                },
+                TaskKind::NextTask => {
+                    // TODO: Anything intelligent to do here?.
```

Comment:
Nice to see this: we can do some code pre-fetching or analysis in the background